token/core/zkatdlog/nogh/loaders.go

```diff
@@ -93,7 +102,15 @@ type VaultPublicParamsLoader struct {
 	PPLabel             string
 }
 
+// Load retrieves the public parameters. It first checks if the public parameters are cached.
+// If not Load fetches the parameters from the ledger
 func (s *VaultPublicParamsLoader) Load() (*crypto.PublicParams, error) {
+	if s.TokenVault == nil {
+		return nil, errors.New("can't load public parameters: please initialize token vault")
```

Comment:
what about adding a constructor and making VaultTokenLoader instead? We can avoid this check every time.

Reply:
For now I will remove this check.